### PR TITLE
Fix for cart rule for carts that have free and paid shipping products

### PR DESCRIPTION
**Pull request by hunter-k**
### Description
The previous code considered only the first item in the cart. Resulting in free shipping be offered when it should not be.

This implementation considers all items in the cart when deciding to offer free shipping.

### Fixed Issues (if relevant)

1. magento/magento2# #12573 Free Shipping Shopping Cart Rule doesn't work with UPS 

### Manual testing scenarios

1. Adding products with paid and unpaid products to cart in various orders.
2. Completing Order

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=16076) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @hunter-k. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on Pull Request changes
- `@magento-engcom-team give me new test instance` - deploy **NEW** test instance based on Pull Request changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy Vanilla Magento instance for Issue or Pull Request

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/OfflineShipping/Model/Quote/Address/FreeShipping.php [hunk 1]
```diff
@@ -65,9 +65,7 @@ public function isFreeShipping(\Magento\Quote\Model\Quote $quote, $items)
             $itemFreeShipping = (bool)$item->getFreeShipping();
             $addressFreeShipping = $addressFreeShipping && $itemFreeShipping;
 
-            if ($addressFreeShipping && !$item->getAddress()->getFreeShipping()) {
-                $item->getAddress()->setFreeShipping(true);
-            }
+            $shippingAddress->setFreeShipping($addressFreeShipping && $itemFreeShipping);
```
Comment (orlangur) [inline app/code/Magento/OfflineShipping/Model/Quote/Address/FreeShipping.php:68]: After the change there are two exact same rows:

```
$addressFreeShipping = $addressFreeShipping && $itemFreeShipping;
 	 
+            $shippingAddress->setFreeShipping($addressFreeShipping && $itemFreeShipping);
```
Why `$item->getAddress()->setFreeShipping(true)` is ignored now?

Please describe steps in more details. As I understand the old logic, it's enough to have one item offering free shipping in order to get free for the address and it looks perfectly valid to me. In new logic it's seems like you try to check ALL items, isn't it?

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
Closing due to inactivity. Feel free to reach me out anytime later if you wish to continue work on this pull request and it will be reopened.

**Comment by markalston** [comment]
@hunter-k  Would you test my PR #14277 which is a more complete fix for this issue.  Let me know if this fixes your problems.  I am trying to get this merged so that I can quit having to hand patch each version update.

**Comment by hunter-k** [comment]
@markalston I can confirm that your changes are working when I tested them.

**Pull request closed without merging**